cylc/flow/scripts/install.py

```diff
@@ -105,6 +105,18 @@ def get_option_parser():
         default=None,
         dest="source")
 
+    parser.add_option(
+        "--symlink-dirs",
+        help=(
+            "Enter a list, in the form 'log=path/to/store, share = $...'"
+            ". Use this option to override local symlinks for directories run,"
+            " log, work, share, share/cycle, as configured in global.cylc. "
+            "Enter an empty list \"\" to skip making localhost symlink dirs."
```

Comment:
```suggestion
            "Enter a list, in the form 'log=path/to/log-dir, share=path/to/share-dir, ...'"
            ". Use this option to override local symlinks for directories run,"
            " log, work, share, share/cycle, as configured in global.cylc. "
            "Paths may contain environment variables.  Any use of this "
            "option overrides all global symlink configuration. "
            "Enter an empty list \"\" to skip making localhost symlink dirs."
```

Also, from testing, if I use a space instead of comma to separate `log=..., share=...` (e.g.),  `cylc install` silently fails to make any symlinks. We should probably support comma or space, or else explicitly fail plain spaces to avoid the parsing complication of filenames containing spaces.

Reply:
And regarding this (my wording):
> Any use of this option overrides all global symlink configuration.

I presume that was a deliberate choice?  What's the argument for not having individual path override?

Reply:
This was a request from @dpmatthews. We discussed did discuss other options here, with the  suggestion from @oliver-sanders that we could add a further option in time, to fill links from config if users wanted it.

Reply:
My reasoning is that this way it's clear to the user what they will get. Otherwise, if I wanted to configure my workflow to use a single filesystem I'd need to either know the global config setup to know what to override or I'd need to set all the possible symlink settings just to be sure. We could support some method of choosing whether to do the individual path override but that seems like unnecessary complexity for an option which we don't expect to be widely used.